### PR TITLE
GCW 3380 and 3381 Support Package Description Translation

### PR DESCRIPTION
**Pull request by bharat619**
### Ticket Link: 
https://jira.crossroads.org.hk/browse/GCW-3380
https://jira.crossroads.org.hk/browse/GCW-3381

### What does this PR do?
Ability to provide guidance on the item description field when creating/editing.

### CODE REVIEW

**Review by abulsayyad123 — APPROVED** [review]

File: app/components/gc-textarea.js [hunk 1]
```diff
@@ -1,23 +1,35 @@
...
     this.hideSiblings();
   },
 
+  hasError() {
+    const value = (this.get("value") || "").trim();
+    if (!value && this.get("require")) {
+      return true;
+    }
+    return false;
+  },
```
Comment (patrixr) [inline app/components/gc-textarea.js:26]:
```suggestion
  hasError() {
    const value = (this.get("value") || "").trim();
    return !value && this.get("require");
  },
```

(not really a change request, just pointing it out)

File: app/controllers/items/detail.js [hunk 2]
```diff
@@ -411,6 +413,37 @@ export default GoodcityController.extend(
...
+      async updateDescription(name, val) {
+        const item = this.get("item");
+        if (item.changedAttributes()[name]) {
+          const params = { package: { [name]: val } };
+          await this.runTask(async () => {
+            await this.get("packageService").updatePackage(item, params);
+          }, ERROR_STRATEGIES.MODAL);
+        }
+        this.send("setShowDescSuggestion", false);
+      },
```
Comment (patrixr) [inline app/controllers/items/detail.js:439]: Those 2 methods seem to be doing pretty much the same thing

```suggestion
      async updateAttribute(prop, val, cb = _.noop) {
        const item = this.get("item");
        if (item.changedAttributes()[prop]) {
          const params = { package: { [prop]: val } };
          await this.runTask(async () => {
            await this.get("packageService").updatePackage(item, params);
            cb();
          }, ERROR_STRATEGIES.MODAL);
        }
      },
      
      updateComment(val) {
         return this.updateAttribute('comment', val);
      },
      
      updateDescription(val) {
         return this.updateAttribute('description', val, () => {
             this.send("setShowDescSuggestion", false);
         });
      },
```

File: app/controllers/items/new.js [hunk 3]
```diff
@@ -64,13 +64,16 @@ export default GoodcityController.extend(
...
+    selectedDescriptionLanguage: "en",
     cancelWarning: t("items.new.cancel_warning"),
     displayFields: Ember.computed("code", function() {
       let subform = this.get("code.subform");
       return this.returnDisplayFields(subform);
     }),
     offersLists: [],
-
+    hasInvalidDescription: Ember.computed("descriptionEn", function() {
+      return !!!this.get("descriptionEn");
```
Comment (patrixr) [inline app/controllers/items/new.js:75]:
```suggestion
      return !this.get("descriptionEn");
```

File: app/controllers/items/detail.js [hunk 4]
```diff
@@ -644,6 +677,23 @@ export default GoodcityController.extend(
...
+      addDefaultDescriptionFor(language) {
+        const item = this.get("item");
+        if (language === "en") {
+          const description = this.get("item.code.descriptionEn");
+          item.set("notes", description);
+        } else {
+          const description = this.get("item.code.descriptionZhTw");
+          item.set("notesZhTw", description);
+        }
+        this.send("saveItem", item);
```
Comment (patrixr) [inline app/controllers/items/detail.js:693]:
```suggestion
        await this.send("saveItem", item);
```

I assume `saveItem` is asynchronous

**Review by swatijadhav — APPROVED** [review]

**Pull request merged**